src/lib.rs

```diff
@@ -361,845 +92,302 @@ impl Navigator {
     }
 
     pub fn init(&mut self) {
-        self.self_test();
-        //Initialize devices on navigator's default settings,
-        //read more on ./navigator-api.pdf
-        self.mag.init().unwrap();
-        self.mag.set_mode(mag_Mode::Cont200Hz).unwrap();
-
-        self.imu
-            .setup(&mut Delay {})
-            .expect("Error: Failed on IMU setup");
-        self.imu.set_accel_range(AccelRange::Range_2g).unwrap();
-        self.imu.set_gyro_range(GyroRange::Range_250dps).unwrap();
-
-        self.adc.reset_internal_driver_state();
-        self.adc
-            .set_full_scale_range(ads1x1x::FullScaleRange::Within4_096V)
-            .unwrap();
-        self.adc
-            .set_data_rate(ads1x1x::DataRate16Bit::Sps860)
-            .unwrap();
-
-        self.pwm.reset_internal_driver_state();
-        self.pwm.use_external_clock().unwrap();
-        self.pwm.set_prescale(100).unwrap();
-        self.pwm.enable().unwrap();
-
-        self.bmp.zero().unwrap();
-
-        self.led.set_led_all(false);
-
-        self.leak
-            .export()
-            .expect("Error: Failed to export leak pin");
-        Delay {}.delay_ms(30_u16);
-        self.leak
-            .set_direction(Direction::In)
-            .expect("Error: Failed to set leak pin as input");
+        // Initialize devices that require it
+        for device in &mut self.devices {
+            if let Some(mag) = device.as_any_mut().downcast_mut::<Ak09915Device>() {
+                mag.init().unwrap();
+            } else if let Some(imu) = device.as_any_mut().downcast_mut::<Icm20689Device>() {
+                imu.init().unwrap();
+            } else if let Some(bmp) = device.as_any_mut().downcast_mut::<Bmp280Device>() {
+                bmp.init().unwrap();
+            } else if let Some(adc) = device.as_any_mut().downcast_mut::<Ads1115Device>() {
+                adc.init().unwrap();
+            }
+            // Initialize other devices as needed
+        }
     }
 
     pub fn self_test(&mut self) -> bool {
```

Comment:
it removed the feature with a false positive

Reply:
What ?

Reply:
fixed